android/app/src/main/java/app/organicmaps/editor/viewmodel/OsmLoginViewModel.java

```diff
@@ -0,0 +1,70 @@
+package app.organicmaps.editor.viewmodel;
+
+import androidx.lifecycle.ViewModel;
+import app.organicmaps.editor.LoginStatusListener;
+import app.organicmaps.sdk.editor.OsmOAuth;
+import app.organicmaps.sdk.util.concurrency.ThreadPool;
+import app.organicmaps.sdk.util.concurrency.UiThread;
+
+public class OsmLoginViewModel extends ViewModel
+{
+  String username;
+  String oauthToken;
+
+  LoginStatusListener listener;
+
+  private UIState uiState = UIState.INITIAL;
+
+  public UIState getUiState()
+  {
+    return uiState;
+  }
+
+  public String getUsername()
+  {
+    return username;
+  }
+
+  public String getOauthToken()
+  {
+    return oauthToken;
+  }
+
+  public void setListener(LoginStatusListener listener)
```

Comment:
```suggestion
  public void setListener(@NonNull LoginStatusListener listener)
```

Reply:
Thanks for the review, Kotlin has made me forget Java 😛 

Addressed: [1cb29c3](https://github.com/organicmaps/organicmaps/pull/11819/commits/1cb29c3a33bcf1830da8df0f5fefbbbeb91304c8)